.github/workflows/build_tvos.yml

```diff
@@ -0,0 +1,154 @@
+# Workflow to handle building the Unity SDK on tvOS
+name: Build tvOS (SubWorkflow)
+
+on:
+  workflow_dispatch:
+  inputs:
+      unity_version:
+        description: 'Unity version'
+        default: '2019'
+        required: true
+        type: string
+      firebase_cpp_sdk_version:
+        description: 'Firebase CPP SDK version to build against (The branch, tag or SHA to checkout)'
+        default: 'v9.6.0'
+        required: false
+        type: string
+      unity_branch:
+        description: 'Unity branch to build against, empty means current branch'
+        default: ''
+        type: string
+      apis:
+        description: 'CSV of apis to build and test'
+        default: 'analytics,auth,crashlytics,database,firestore,functions,installations,messaging,remote_config,storage'
+        required: true
+        type: string
+      unity_platform_name:
+        description: 'The platform name Unity should install with'
+        default: 'tvOS'
+        required: true
+        type: string
+
+  workflow_call:
+    inputs:
+      unity_version:
+        description: 'Unity version'
+        default: '2019'
+        required: true
+        type: string
+      firebase_cpp_sdk_version:
+        description: 'Firebase CPP SDK version to build against (The branch, tag or SHA to checkout)'
+        default: ''
+        required: false
+        type: string
+      unity_branch:
+        description: 'Unity branch to build against, empty means current branch'
+        default: ''
+        type: string
+      apis:
+        description: 'CSV of apis to build and test'
+        default: 'analytics,auth,crashlytics,database,firestore,functions,installations,messaging,remote_config,storage'
+        required: true
+        type: string
+      unity_platform_name:
+        description: 'The platform name Unity should install with'
+        default: ''
+        required: true
+        type: string
+
+jobs:
+  build_tvos:
+    name: build-tvos-unity${{ inputs.unity_version }}-CPP${{ inputs.firebase_cpp_sdk_version }}
```

Comment:
echo to the previous comment, this would not take effect if the workflow start with dispatch.

Reply:
Confused by this comment. Both the dispatch and the trigger have these inputs, no?

Reply:
My experience is the workflow_dispatch use another parameter name for the input, needs to be events.inputs something like that. workflow_call is using "inputs" directly. examples can be found build_starter.yml, - id: set_outputs step.

Reply:
I think it should be ok, based on this reference:
https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#onworkflow_dispatchinputs